snakemake/io.py

```diff
@@ -172,7 +172,7 @@ async def worker(queue):
         await asyncio.gather(*tasks)
 
     async def collect_mtime(self, path):
-        return path.mtime_uncached
+        return path.mtime
```

Comment:
Let us not change this here. I have just created a separate PR with a more efficient change that in this special case involves fewer lookups: #2284.

```suggestion
        return path.mtime_uncached
```